src/AWS.Deploy.Orchestration/SystemCapabilityEvaluator.cs

```diff
@@ -27,105 +28,161 @@ public class SystemCapabilityEvaluator : ISystemCapabilityEvaluator
+                return new DockerInfo(false, "");
+            }
         }
 
         /// <summary>
         /// From https://docs.aws.amazon.com/cdk/latest/guide/work-with.html#work-with-prerequisites,
         /// min version is 10.3
         /// </summary>
-        private async Task<Version?> GetNodeJsVersion()
+        private async Task<NodeInfo> GetNodeJsVersionAsync()
         {
-            // run node --version to get the version
-            var result = await _commandLineWrapper.TryRunWithResult("node --version");
+            var cancellationTokenSource = new CancellationTokenSource();
+            cancellationTokenSource.CancelAfter(CAPABILITY_EVALUATION_TIMEOUT_MS);
+
+            try
+            {
+                // run node --version to get the version
+                var result = await _commandLineWrapper.TryRunWithResult("node --version", cancellationToken: cancellationTokenSource.Token);
+
+                var versionString = result.StandardOut ?? "";
 
-            var versionString = result.StandardOut ?? "";
+                if (versionString.StartsWith("v", StringComparison.OrdinalIgnoreCase))
+                    versionString = versionString.Substring(1, versionString.Length - 1);
 
-            if (versionString.StartsWith("v", StringComparison.OrdinalIgnoreCase))
-                versionString = versionString.Substring(1, versionString.Length - 1);
+                if (!result.Success || !Version.TryParse(versionString, out var version))
+                    return new NodeInfo(null);
 
-            if (!result.Success || !Version.TryParse(versionString, out var version))
-                return null;
+                return new NodeInfo(version);
 
-            return version;
+            }
+            catch (TaskCanceledException)
+            {
+                // If the check timed out, treat Node as not installed
+                return new NodeInfo(null);
+            }
         }
 
         /// <summary>
         /// Checks if the system meets all the necessary requirements for deployment.
         /// </summary>
         public async Task<List<SystemCapability>> EvaluateSystemCapabilities(Recommendation selectedRecommendation)
         {
-            var capabilities = new List<SystemCapability>();
-            var systemCapabilities = await Evaluate();
+            var missingCapabilitiesForRecipe = new List<SystemCapability>();
             string? message;
+
+            // We only need to check that Node is installed if the user is deploying a recipe that uses CDK
             if (selectedRecommendation.Recipe.DeploymentType == DeploymentTypes.CdkProject)
             {
-                if (systemCapabilities.NodeJsVersion == null)
+                // If we haven't cached that NodeJS installation is valid, or the cache is expired
+                if (_nodeDependencyValidUntilUtc == null || DateTime.UtcNow >= _nodeDependencyValidUntilUtc)
```

Comment:
nit: Consider setting the default value of `_nodeDependencyValidUntilUtc` to [DateTime.MinValue](https://learn.microsoft.com/en-us/dotnet/api/system.datetime.minvalue?view=net-8.0) instead on null.

Reply:
Done in https://github.com/aws/aws-dotnet-deploy/pull/837/commits/b22ed89708bf1b4e05ec4b89d774f841be431331